methods.py

```diff
@@ -863,12 +863,16 @@ def clean_cache(cache_path: str, cache_limit: int, verbose: bool):
     texts = []
     stats = []
     for file in files:
+        # Save file stats to rewrite after modifying.
+        tmp_stat = os.stat(file)
         # Failing a utf-8 decode is the easiest way to determine if a file is binary.
         try:
             with open(file, encoding="utf-8") as out:
                 out.read(1024)
         except UnicodeDecodeError:
-            stats.append((file, *os.stat(file)[6:8]))
+            stats.append((file, *tmp_stat[6:8]))
+            # Restore file stats after reading.
+            os.utime(file, (tmp_stat[7], tmp_stat[8]))
```

Comment:
Why is atime restored only when exception raised?

Reply:
Because the exception is used to filter files, please see the above comment, text files are deleted unconditionally so no need to restore their stats

Reply:
1. I meant why it is only for this exception and not for OSError
2. I think I understood why... OSError should mean there is no atime update...
3. I am too spamming today :woman_facepalming:  Sorry